src/vector/backends/awkward.py

```diff
@@ -1978,16 +1978,25 @@ def impl(record: typing.Any) -> typing.Any:
     return context.compile_internal(builder, impl, sig, args)
 
 
-ak.behavior["__numba_typer__", "Vector2D"] = _numba_typer_Vector2D
-ak.behavior["__numba_typer__", "Vector3D"] = _numba_typer_Vector3D
-ak.behavior["__numba_typer__", "Vector4D"] = _numba_typer_Vector4D
-ak.behavior["__numba_typer__", "Momentum2D"] = _numba_typer_Momentum2D
-ak.behavior["__numba_typer__", "Momentum3D"] = _numba_typer_Momentum3D
-ak.behavior["__numba_typer__", "Momentum4D"] = _numba_typer_Momentum4D
-
-ak.behavior["__numba_lower__", "Vector2D"] = _numba_lower
-ak.behavior["__numba_lower__", "Vector3D"] = _numba_lower
-ak.behavior["__numba_lower__", "Vector4D"] = _numba_lower
-ak.behavior["__numba_lower__", "Momentum2D"] = _numba_lower
-ak.behavior["__numba_lower__", "Momentum3D"] = _numba_lower
-ak.behavior["__numba_lower__", "Momentum4D"] = _numba_lower
+behavior["__numba_typer__", "Vector2D"] = _numba_typer_Vector2D
+behavior["__numba_typer__", "Vector3D"] = _numba_typer_Vector3D
+behavior["__numba_typer__", "Vector4D"] = _numba_typer_Vector4D
+behavior["__numba_typer__", "Momentum2D"] = _numba_typer_Momentum2D
+behavior["__numba_typer__", "Momentum3D"] = _numba_typer_Momentum3D
+behavior["__numba_typer__", "Momentum4D"] = _numba_typer_Momentum4D
+
+behavior["__numba_lower__", "Vector2D"] = _numba_lower
+behavior["__numba_lower__", "Vector3D"] = _numba_lower
+behavior["__numba_lower__", "Vector4D"] = _numba_lower
+behavior["__numba_lower__", "Momentum2D"] = _numba_lower
+behavior["__numba_lower__", "Momentum3D"] = _numba_lower
+behavior["__numba_lower__", "Momentum4D"] = _numba_lower
```

Comment:
Was this using the global behavior for a reason?

Reply:
There's a one `vector.backends.awkward.behavior` dict, but this doesn't get installed in the global `ak.behavior` unless the user opts in (calls `vector.register_awkward()`). Vector's `behavior` dict can be passed around between arrays, so that it applies to some arrays and not others. Just filling up Vector's `behavior` dict is not something I'd call "global."

That said, I'm more inclined toward recommending globally installing behaviors with `vector.register_awkward()` than I used to be. It's global state, but it's assigned once at the beginning of a typical script/library and not changed thereafter. Class and function definitions are also "global state" in this sense, and behaviors are more like class/function definitions than application data. (If it were a compiled language, this would all be precompiled code.) Having this globally installed is better than getting confused about which arrays have Vector behaviors and which don't.

Reply:
I'm on mobile right now, but originally this code was directly using ak.behaviour, which is what i was referring to. It sounds like it wasn't supposed to, based on your reply!

Reply:
I just noticed the red: you're right! That was a giant mistake. It's not one that anyone would have noticed, but still. These functions are supposed to go into Vector's `behavior` dict, not Awkward's.